app/helpers/loss_report_helper.rb

```diff
@@ -0,0 +1,211 @@
+    # this method would return /Contacts/Contact because Contact is the outermost array (or false if the path doesn't contain an array)
+    pre_translation_array, pre_translation_path = hash_navigation(path, orig_h)
+    post_translation_array, post_translation_path = hash_navigation(path, conv_h)
+
+    return false if pre_translation_array == false && post_translation_array == false
+    return pre_translation_path if pre_translation_array.is_a?(Array)
+    return post_translation_path if post_translation_array.is_a?(Array)
+
+    # the number of keys must be 1 because all arrays in echo10, dif10, and iso19115 are tagged similar to:
+    # <Contacts><Contact>contact</Contact></Contacts> and so all array-containing tags will be the plural
+    # of the array name. This clause serves to identify array-containing tags when their paths aren't properly
+    # displayed by nokogiri
+    if pre_translation_array.is_a?(Hash) && pre_translation_array.keys.length == 1 && pre_translation_array[pre_translation_array.keys[0]].is_a?(Array)
+      return "#{pre_translation_path}/#{pre_translation_array.keys[0]}"
+    elsif post_translation_array.is_a?(Hash) && post_translation_array.keys.length == 1 && post_translation_array[post_translation_array.keys[0]].is_a?(Array)
+      return "#{post_translation_path}/#{post_translation_array.keys[0]}"
+    end
+    path_contains_array = false
+  end
+
+  def add_to_report(change, element, path, hide_items, disp, json_output, text_output)
+    @counter ||= 0 and @counter += 1
+    # this function serves to preclude complex nests from forming in loss_report_output the
+    # following 'if' structure is intended to increase readability by eliminating nests
+    return text_output.concat("#{@counter}.".ljust(4)+"#{change}: #{element}".ljust(60) + path + "\n") if hide_items == false && disp == 'text'
+    return text_output.concat("#{@counter}.".ljust(4)+"#{change}: ".ljust(3) + path + "\n") if hide_items == true && disp == 'text'
+    return json_output["#{@counter}. #{change}: #{path}"] = element if disp == 'json'
+  end
+
+  def hash_map(hash)
+    buckets = Array.new
+    hash.each do |key,val|
+      if val.is_a? Hash then hash_map(val).each do |item|
+        item['path'] = key + '/' + item['path']
+        buckets << item end
+      else
+        buckets << {'path'=> key, 'value'=> val}
+      end
+    end
+    buckets
+  end
+
+  def prepare_collections(concept_id, umm_c_version)
+    original_collection_native_xml = cmr_client.get_concept(concept_id,token, {})
+    return false if !original_collection_native_xml.success?
+
+    content_type = original_collection_native_xml.headers.fetch('content-type').split(';')[0]
+    original_collection_native_hash = Hash.from_xml(original_collection_native_xml.body)
+    translated_collection_umm_json = cmr_client.translate_collection(original_collection_native_xml.body, content_type, "application/vnd.nasa.cmr.umm+json;version=#{umm_c_version}", skip_validation=true)
```

Comment:
The `params[format:]` yields something like `text` or `json`.  Is there another `params[<symbol>]` that will yield the info needed?

Reply:
`before_action :set_collection`
means that the set_collection method is being called before each action.  Anything set there should be accessible in the helper when it is called, including @revisions.  I believe that it should be an array of revisions where the first is the most recent, but you should verify that before using it.

Reply:
Ohh, I thought you were talking about the `content_type`, I see you're talking about the whole collection- will fix.

Reply:
Ah, so it seems that the concept obtained within `set_collection` (the variable is `@collection`) is of the umm-c format so I would still have to `get_concept` in the `prepare_collections` method. Please let me know if I'm misunderstanding something.

Reply:
You need to fetch the native (before) and get the back-translation (after). but you don't need to translate the before to the umm-json because you already have it.

Reply:
I'm not exactly sure what the difference would be between asking cmr for 1. the UMM version of a collection or 2. the native version of a collection and then using the translate endpoint to convert to UMM. That being said, is it possible that the UMM collection that is retrieved by `set_collections` could be different to the one that is produced by retrieving the native collection and then translating it to UMM? I tested a few collections and haven't found any differences in the loss reports but I imagine it _might_ be possible for those outputs to be different. You may be able to speak more to this..

Reply:
`That being said, is it possible that the UMM collection that is retrieved by set_collections could be different to the one that is produced by retrieving the native collection and then translating it to UMM?`
Erich says that the translation endpoint uses the same code as their internal translation from native to umm-json.  Since the set collections is already being run, you can save on that translation.

Reply:
perfect. Sounds good.